src/crimson/os/seastore/cached_extent.h

```diff
@@ -549,7 +549,12 @@ class CachedExtent
   /// Return true if extent is fully loaded or is about to be fully loaded (call 
   /// wait_io() in this case)
   bool is_fully_loaded() const {
-    return ptr.has_value();
+    if (ptr.has_value()) {
+      // length == 0 iff root
```

Comment:
I'm a little confused, is the comment misplaced? And why changing the code not to directly `return ptr.has_value()`?

Seems that this change should be in the following `get_loaded_length()`, am I right?

Reply:
> Seems that this change should be in the following get_loaded_length(), am I right?

This is splitted into 2 commits, the 1st commit "clarify invarients in the cstors" only do cleanups that should not touch the logic.

The 2st commit "implement partial reads..." added necessary asserts for the feature.

> I'm a little confused, is the comment misplaced?

No, if it is root, it always set a valid (but empty) ptr, see below, because root won't be partially loaded. And if ptr is invalid (i.e. partially loaded), it must not be root, so length won't be 0.

> And why changing the code not to directly return ptr.has_value()?

Just wanted to add asserts to explain the invarients more clearly.